js/src/common/components/ColorPreviewInput.tsx

```diff
@@ -0,0 +1,26 @@
+import type Mithril from 'mithril';
+
+import Component, { ComponentAttrs } from '../Component';
+import classList from '../utils/classList';
+import icon from '../helpers/icon';
+
+export default class ColorPreviewInput extends Component {
+  value?: string;
+
+  view(vnode: Mithril.Vnode<ComponentAttrs, this>) {
+    const { className, ...attrs } = this.attrs;
+    const value = attrs.bidi?.() || attrs.value;
+
+    attrs.type ||= 'text';
+
+    return (
+      <div className="ColorInput">
+        <input className={classList('FormControl', className)} {...attrs} />
+
+        <span className="ColorInput-icon">{icon('fas fa-exclamation-circle')}</span>
+
+        <div className="ColorInput-preview" style={{ '--input-value': value }} />
+      </div>
```

Comment:
One topic of note here is accessibility.

I didn't think about this earlier, but we would want some way of alerting a screenreader to the fact that the input is invalid.

A11Y isn't a massive one of our focuses on the admin panel, so I'm fine with not implementing this in the time being, but some aria roles to indicate element metadata would be nice:

```suggestion
      <div className="ColorInput">
        <input className={classList('FormControl', className)} {...attrs} />

        <span className="ColorInput-icon" role="presentation">{icon('fas fa-exclamation-circle')}</span>

        <div className="ColorInput-preview" style={{ '--input-value': value }} role="presentation" />
      </div>
```